pages/documents/DeveloperTools/FaaS/Developing/toolbelt.md

```diff
@@ -377,3 +415,163 @@ conversationUtil.getConversationById(conversationId)
   .then(replacedFiles => //TODO: react on the response)
   .catch(err => //TODO: React to error);
 ```
+### SDE Util
+
+The SDE Util allows to perform SDE related methods, which are listed below. This Util works for **Messaging-Use Cases only**!
+
+#### Add SDEs
+This method adds/ updates SDEs to an Engagement via the [Engagement Attributes API](https://developers.liveperson.com/engagement-attributes-api-overview.html). It requires the session ID and the visitor ID where the SDEs should be added and the SDEs themselves. It returns a `Promise` that resolves when the SDEs have been successfuly set/ updated.
+
+**Parameters**
+
+<table style="width: 100%;">
+<thead>
+<tr>
+<th>Parameter</th>
+<th>Required?</th>
+<th>Description</th>
+<th>Type/Value</th>
+</tr>
+</thead>
+<col width="20%">
+<col width="15%">
+<col width="50%">
+<col width="15%">
+<tbody>
+<tr>
+<td>sdes</td>
+<td>yes</td>
+<td>Array of SDEs which should be added to an engagement. For more information about the available Engagement Attributes, refer to the <a href="https://developers.liveperson.com/engagement-attributes-overview.html">Engagement Attributes Overview</a>.</td>
+<td>array</td>
+</tr>
+<tr>
+<td>visitorId</td>
+<td>yes</td>
+<td>The ID of the Visitor you want to add the SDEs to</td>
+<td>string</td>
+</tr>
+<tr>
+<td>sessionId</td>
+<td>yes</td>
+<td>The ID of the Session you want to add the SDEs to</td>
+<td>string</td>
+</tr>
+</tbody>
+</table>
+
+
+
+**Sample Usage**
+
+```javascript
+  // import FaaS Toolbelt
+  const { Toolbelt } = require("lp-faas-toolbelt");
+
+  // Create instance
+  const sdeUtil = Toolbelt.SDEUtil();
+
+  // Define parameters
+  const visitorId = "XXXXXX";
```

Comment:
Maybe mention that you can get these Ids in the messaging related events

Reply:
implemented